powerstore/resource_snapshotrule.go

```diff
@@ -0,0 +1,254 @@
+				MarkdownDescription: "The time of the day to take a daily snapshot, with format hh:mm.",
+			},
+			"timezone": {
+				Type:                types.StringType,
+				Optional:            true,
+				Computed:            true,
+				Description:         "The time zone identifier for applying the time zone to the time_of_day for a snapshot rule.",
+				MarkdownDescription: "The time zone identifier for applying the time zone to the time_of_day for a snapshot rule.",
+			},
+			"days_of_week": {
+				Type:                types.ListType{ElemType: types.StringType},
+				Optional:            true,
+				Computed:            true,
+				Description:         "The days of the week when the snapshot rule should be applied.",
+				MarkdownDescription: "The days of the week when the snapshot rule should be applied.",
+			},
+			"desired_retention": {
+				Type:                types.Int64Type,
+				Required:            true,
+				Description:         "The Desired snapshot retention period in hours to retain snapshots for this time period.",
+				MarkdownDescription: "The Desired snapshot retention period in hours to retain snapshots for this time period.",
+			},
+			"is_replica": {
+				Type:                types.BoolType,
+				Computed:            true,
+				Description:         "Indicates whether this is a replica of a snapshot rule on a remote system.",
+				MarkdownDescription: "Indicates whether this is a replica of a snapshot rule on a remote system.",
+			},
+			"nas_access_type": {
+				Type:                types.StringType,
+				Optional:            true,
+				Computed:            true,
+				Description:         "The NAS filesystem snapshot access method for snapshot rule.",
+				MarkdownDescription: "The NAS filesystem snapshot access method for snapshot rule.",
+			},
+			"is_read_only": {
+				Type:                types.BoolType,
+				Optional:            true,
+				Computed:            true,
+				Description:         "Indicates whether this snapshot rule can be modified.",
+				MarkdownDescription: "Indicates whether this snapshot rule can be modified.",
+			},
+			"managed_by": {
+				Type:                types.StringType,
+				Computed:            true,
+				Description:         "The entity that owns and manages the instance.",
+				MarkdownDescription: "The entity that owns and manages the instance.",
+			},
+			"managed_by_id": {
+				Type:                types.StringType,
+				Computed:            true,
+				Description:         "The unique id of the managing entity.",
+				MarkdownDescription: "The unique id of the managing entity.",
+			},
+		},
+	}, nil
+}
+
+// NewResource is a wrapper around provider
+func (r resourceSnapshotRuleType) NewResource(_ context.Context, p tfsdk.Provider) (tfsdk.Resource, diag.Diagnostics) {
+	return resourceSnapshot{
+		p: *(p.(*Pstoreprovider)),
+	}, nil
+}
+
+type resourceSnapshot struct {
```

Comment:
resourceSnapshotRule will be more relevant

Reply:
Yeah, fixing it